notebooker/execute_notebook.py

```diff
@@ -263,6 +263,16 @@ def _get_overrides(overrides_as_json: AnyStr, iterate_override_values_of: Option
     return all_overrides
 
 
+def env_coupled_var(var_value: Optional[str], env_name: str) -> Optional[str]:
+    """Coalesce a value from the given one and environment, then update the environment."""
+    value = var_value if var_value is not None else os.environ.get(env_name)
+    if value is None:
+        os.environ.pop(env_name, None)
```

Comment:
Is `.pop()` the best choice here? What if e.g. someone is running:

```
export MONGO_HOST=blah
execute_notebook
execute_notebook
```

Reply:
It turns out this branch of the `if` isn't needed, as it will only happen when var_value is None and there the environ isn't set. I just pushed a change to clean it up.

p.s. was it environment isolation you were worried about:
```sh
export MONGO_HOST=example1
# this will use example2 and update its copy of the environment to MONGO_HOST=example2
execute_notebook --mongo-host=example2
# this will use example1 as its copy of the environment is a copy from the shell
execute_notebook
```